modules/hal_nordic/nrfx/nrfx_usbd.c

```diff
@@ -0,0 +1,1892 @@
+/*
```

Comment:
Is there a reason why you would just not fold this directly into https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/usb/device/usb_dc_nrfx.c? What's the point of keeping two separate files?

Reply:
This is also used by the new stack (drivers/usb/udc/udc_nrf.c). The changes done here benefit for the old and new stack. I prefer to first simplify the code and only then do the merge.

Reply:
OK, makes sense.